index.bs

```diff
@@ -450,8 +450,27 @@ To <dfn>pullDatagrams</dfn>, given a {{WebTransport}} object |transport|, run th
   1. Set |datagrams|.{{[[IncomingDatagramsPullPromise]]}} to a new promise.
   1. Return |datagrams|.{{[[IncomingDatagramsPullPromise]]}}.
 1. Let |bytes| and |timestamp| be the result of [=dequeuing=] |queue|.
-1. Let |chunk| be a new {{Uint8Array}} object representing |bytes|.
-1. [=ReadableStream/Enqueue=] |chunk| to |transport|.{{[[Datagrams]]}}.{{WebTransportDatagramDuplexStream/[[Readable]]}}.
+1. Let |buffer|, |offset|, |maxBytes|, and |chunk| be null.
+1. If |datagrams|.{{WebTransportDatagramDuplexStream/[[Readable]]}}'s
+   [=ReadableStream/current BYOB request view=] is not null, then:
+  1. Let |view| be |datagrams|.{{WebTransportDatagramDuplexStream/[[Readable]]}}'s
+     [=ReadableStream/current BYOB request view=].
+  1. If |view|'s [=BufferSource/byte length=] is less than the size of |bytes|,
+     [=ReadableStream/error=] |datagrams|.{{WebTransportDatagramDuplexStream/[[Readable]]}}
+     with a {{RangeError}}.
+  1. Otherwise:
+    1. Set |offset| to |view|.\[[ByteOffset]].
+    1. Set |maxBytes| to |view|'s [=BufferSource/byte length=].
+    1. Set |buffer| to |view|'s [=BufferSource/underlying buffer=].
+    1. Set |chunk| to |view|.
+1. Otherwise:
+  1. Set |offset| to 0.
+  1. Set |maxBytes| to an [=implementation-defined=] size.
```

Comment:
I think the size should probably just be |bytes|, unless there's a reason why some implementations would need it to be larger.